rust-wrapper/src/fft.rs

```diff
@@ -0,0 +1,40 @@
+use ark_bls12_381::Fr as ScalarField;
+use ark_ff::PrimeField;
+use ark_poly::univariate::DensePolynomial;
+use ark_poly::DenseUVPolynomial;
+use libc;
+use num_bigint::BigUint;
+use std::ops::Mul;
+
+use crate::utils::deserialize_vector_scalar_field;
+
+///
+/// # Safety
+/// The caller must ensure that valid pointers and sizes are passed. 
+/// To find out how data should be represented in memory, you can look at fft_bench.rs
```

Comment:
You can extract the safe part of a function which turns slice of bytes into polynomials and returns vector of bytes, and use the safe part in benchmarks, while leaving the unsafe parts to FFI

Reply:
Also I guess there should be a no-op (unsafe) conversion from the (pointer, size) pair into a Rust slice, and it's better to use them